ledger/complete/mtrie/trie/trie.go

```diff
@@ -313,21 +291,37 @@ func (mt *MTrie) proofs(head *node.Node, paths []ledger.Path, proofs []*ledger.T
 	for _, p := range proofs {
 		p.Steps++
 	}
-	// split paths based on the value of i-th bit (i = trie height - node height)
-	lpaths, lproofs, rpaths, rproofs := utils.SplitTrieProofsByPath(paths, proofs, mt.height-head.Height())
+
+	// partition step to quick sort the paths:
+	// lpaths contains all paths that have `0` at the partitionIndex
+	// rpaths contains all paths that have `1` at the partitionIndex
+	partitionIndex := utils.SplitTrieProofsByPath(paths, proofs, mt.Height()-head.Height())
```

Comment:
maybe we could give `mt.Height()-head.Height()` a name, I'm been used in 3 places within the `proof` function